site/en/docs/extensions/reference/printing/index.md

```diff
@@ -2,4 +2,95 @@
 api: printing
 ---
 
-<!-- Intentionally blank -->
+## Manifest
+
+All `chrome.printing` methods and events require you to declare the "printing" permission in the [extension manifest](/docs/extensions/mv3/manifest/). For example:
+
+```json
+{
+  "name": "My extension",
+  ...
+  "permissions": [
+    "printing"
+  ],
+  ...
+}
+```
+
+## Examples
+
+The examples below demonstrate using each of the methods in the printing namespace. This code is copied from or based on the [printing](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/api-samples/printing) in the Extension samples database.
+
+### cancelJob()
+
+This example uses the `onJobStatusChanged` handler to hide a 'cancel' button when the `jobStatus` is neither `PENDING` or `IN_PROGRESS`. Note that on some networks or when a Chromebook is connected directly to the printer, these states may pass too quickly for the cancel button to be visible long enough to be called. This is greatly simplified the printing example.
+
+```javascript
+chrome.printing.onJobStatusChanged.addListener((jobId, status) => {
+  const cancelButton = document.getElementById("cancelButton");
+  cancelButton.addEventListener('click', () => {
+    chrome.printing.cancelJob(jobId).then((response) => {
+      if (response !== undefined) {
+        console.log(response.status);
+      }
+      if (chrome.runtime.lastError !== undefined) {
+        console.log(chrome.runtime.lastError.message);
+      }
+    });
+  });
+  if (status !== "PENDDING" && status !== "IN_PROGRESS") {
+    cancelButton.style.visibility = 'hidden';
+  } else {
+    cancelButton.style.visibility = 'visible';
+  }
+}
+```
+
+### getPrinters() and getPrinterInfo()
+
+A single example is used for these functions because getting printer information requires a printer ID, which is retrieved by calling `getPrinters()`. This example logs the name and description of the default printer to the console. This is a simplified version of the printing example.
+
+```javascript/0,2
+​​const printers = await chrome.printing.getPrinters();
+const defaultPrinter = printers.find((printer) => {
+  const printerInfo = await chrome.printing.getPrinterInfo(printer.id);
+  return printerInfo.isDefault;
+}
+console.log(`Default printer: ${defaultPrinter.name}.\n\t${defaultPrinter.description}`);
+```
+
+### submitJob()
+
+The `submitJob()` method requires three things.
+* A `ticket` structure specifying which capabilities of the printer are to be used. If the user needs to select from available capabilities, you can retrieve them for a specific printer using `getPrinterInfo()`.
```

Comment:
> a specific printer using `getPrinterInfo()`.
How does this work? Should we add a link?

Also in the code below says:
> const ticket = getPrinterTicket(printerId);

Reply:
I was trying to avoid making this overly complicated. I think I threaded the needle better. Please take a look.